enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,362 @@
+
+#### 1. Calling spoke cluster's kubernetes api 
+
+## Proposal
+
+#### Architecture
+
+
+The following figure shows the interaction between the components:
+
+```
+   
+                     -----------------------------------------
+                     | Hub-Cluster                           |
+                     |                                       |
+  ----------------   | ----------------     ---------------  |
+  | Proxy Client |> > >| Proxy Server | <===| Proxy Addon |  |
+  ----------------   | ----------------     |  Controller |  |
+                     |      V               ---------------  |
+                     |      V                    ‖           |
+                     |    / V \                  ‖           |
+                     ----/| V |\-----------------‖------------
+                          | V |                  ‖
+                   Reverse| V |Tunnel            ‖ Install/Operate
+                          | V |                  ‖ 
+ -------------------------| V |-------------     ‖
+ | Managed-Cluster        | V |            |     ‖
+ |                        | v |            |     ‖
+ | -----------------   ---------------     |     ‖
+ | | KubeApiserver |< <| Proxy Agent | <==========
+ | -----------------   ---------------     |
+ -------------------------------------------
+
+```
+
+- A proxy-agent that establishes a long connection towards the proxy-server
+  and register itself w/ the cluster's identity.
+- A proxy-server that receives handshakes from proxy-agent and maintains 
+  tunnel over the long connection.
+- A proxy-client requesting the target kube-apiserver through the tunnel
+  over either websocket HTTP-CONNECT or grpc protocol.
+- An addon controller automates the installation of proxy servers and proxy 
+  agents by consuming the corresponding configuration custom resource. And
+  optionally the proxy-agents can aggregate and report the status upwards
+  so that admins can checkout their healthiness easily w/o even touching
+  the managed-clusters.
+
+
+#### Components
+
+##### Cluster Proxy Operator
+
+According to the latest design and implementation of [addon-framework](https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/8-addon-framework),
+we will bring a new agent controller named `cluster-proxy-operator` that
+manages the lifecycle of the proxy components by watching the configuration
+under a new group of api `proxy.addon.open-cluster-management.io/v1alpha1`:
+
+- __ManagedProxyServer__: the configuration and status of proxy-servers.
+
+- __MangedProxyAgent__: the configuration and status of proxy-agents.
+
+Additionally, the operator will be list-watching the standard `ManagedClusterAddon`
+resource and automatically installs proxy-agents w/ the expected options
+upon discovering the addon in each cluster's namespace.
+
+
+##### ManagedProxyServer
+
+The `ManagedProxyServer` is supposed to be a cluster-scoped resource and holds 
```

Comment:
Why do we need it to be a cluster scoped resource or singleton? I think it is possible that we could have multiple servers,  each with a ManagedClusterSet also?

Reply:
can you elaborate how to make it a singleton in k8s api? practically i think singleton should be sufficient, but having forward-compatibility of multiple groups of servers isnt a bad idea, is it? i can imagine cases where we need to isolate a new tunnel so that protecting some traffic for critical usage from bandwidth congestion